languages/csharp/exercises/concept/structs/.meta/Example.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Collections.Generic;
+
+public struct Coord
+{
+    public Coord(ushort x, ushort y)
+    {
+        X = x;
+        Y = y;
+    }
+
+    public int X {get; }
+    public ushort Y {get; }
+
+    public bool Equals(Coord other)
+    {
+        return X == other.X && Y == other.Y;
+    }
+
+    public override bool Equals(object obj)
+    {
+        return obj is Coord other && Equals(other);
+    }
+
+    public override int GetHashCode()
+    {
+        return HashCode.Combine(X, Y);
+    }
+}
+
+public struct Plot
+{
+    public Plot(Coord topLeft, Coord topRight, Coord bottomLeft, Coord bottomRight)
+    {
+        TopLeft = topLeft;
+        TopRight = topRight;
+        BottomLeft = bottomLeft;
+        BottomRight = bottomRight;
+        str = "mike";
+    }
+
+    public Coord TopLeft {get; }
+    public Coord TopRight {get; }
+    public string str { get; }
+    public Coord BottomLeft {get; }
+    public Coord BottomRight {get; }
+
+    public bool Equals(Plot other)
+    {
+        return TopLeft.Equals(other.TopLeft) 
+               && TopRight.Equals(other.TopRight) 
+               && BottomLeft.Equals(other.BottomLeft) 
+               && BottomRight.Equals(other.BottomRight);
+    }
+
+    public override bool Equals(object obj)
+    {
+        return obj is Plot other && Equals(other);
+    }
+
+    public override int GetHashCode()
+    {
+        return HashCode.Combine(TopLeft, TopRight, BottomLeft, BottomRight);
+    }
```

Comment:
This can be removed, right? I assume was for a performance test?

Reply:
I  was going to leave it in.  The performance improvement over no custom equality test was about 20% (tightlyPacked).  Obviously it would not be in the template code.  Do you have better information or more insights?

Reply:
The example should be the idiomatic solution and will be presented to mentors. I would argue that the version without the custom equality is the idiomatic one, but I do get that it is nice to be able to show the performant version. @iHiD I'm not sure where we would put this type of information (a variation of the idiomatic solution of a concept exercise).

Reply:
I have removed the custom `Equals()` and `GetHashCode()` partly because when I first coded the example I did not have them anyway.  

I added them when I read the [equality article](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type) in the docs.

The article does not really provide clear criteria on when to implement a custom `Equals()`.  In this particular example, I was (at least in my own mind) using performance as a reason for using structs which would point to custom routines.  On the other hand the performance difference is only 20% and it is arguable that this constitutes premature optimization.

In the end I am guided by the principle that "the best code is less code".